modules/search-pipeline-common/src/main/java/org/opensearch/search/pipeline/common/ScriptRequestProcessor.java

```diff
@@ -93,10 +89,33 @@ public SearchRequest processRequest(SearchRequest request) throws Exception {
             searchScript = precompiledSearchScript;
         }
         // execute the script with the search request in context
-        searchScript.execute(Map.of("_source", new SearchRequestMap(request)));
+        searchScript.execute(Map.of("_source", new SearchRequestMap(request), "request_context", new RequestContextMap(requestContext)));
         return request;
     }
 
+    private static class RequestContextMap extends BasicMap {
+        private final PipelineProcessingContext pipelinedRequestContext;
+
+        private RequestContextMap(PipelineProcessingContext pipelinedRequestContext) {
+            this.pipelinedRequestContext = pipelinedRequestContext;
+        }
+
+        @Override
+        public Object get(Object key) {
```

Comment:
If I recall correctly, Painless script context is generally passed as a Map. In this case, we're adapting our custom class to the Map interface.

Reply:
Oh I see, than we could just go with `AbstractMap`, right?

Reply:
> I'm happy to make that change if we think it would be better than the `BasicMap` approach.

I see the picture now, thanks @msfroh, I thin we could keep it like that but reshuffle classes a bit:
 - move `BasicMap` to `org.opensearch.search.pipeline.common` and make it package private
 - move the `SearchRequestMap` to `org.opensearch.search.pipeline.common` and make it package private as well

What do you think?